test/e2e/targets/googlecloudstorage/main.go

```diff
@@ -0,0 +1,259 @@
+	})
+
+	Context("a target is deployed", func() {
+		var trgtURL *url.URL
+		var storageClient *storage.Client
+
+		var sentEvent *cloudevents.Event
+
+		var serviceaccountKey string
+		var gcloudProject string
+
+		BeforeEach(func() {
+			var err error
+			serviceaccountKey = e2egcloud.ServiceAccountKeyFromEnv()
+			gcloudProject = e2egcloud.ProjectNameFromEnv()
+
+			storageClient, err = storage.NewClient(context.Background(), option.WithCredentialsJSON([]byte(serviceaccountKey)))
+			Expect(err).ToNot(HaveOccurred())
+
+			gcpSecret = createGCPCredsSecret(f.KubeClient, ns, serviceaccountKey)
+
+			By("creating a Google Cloud Storage Bucket", func() {
+				bucketName = e2estorage.CreateBucket(storageClient, gcloudProject, f)
+
+				DeferCleanup(func() {
+					By("deleting Google Cloud Storage Bucket "+bucketName, func() {
+						e2estorage.DeleteObjects(storageClient, bucketName)
+						e2estorage.DeleteBucket(storageClient, bucketName)
+					})
+				})
+			})
+		})
+
+		When("the spec contains default settings", func() {
+			BeforeEach(func() {
+				By("creating an GoogleCloudStorageTarget object", func() {
+					trgt, err := createTarget(trgtClient, ns, "test-",
+						withBucketName(bucketName),
+						withCredentials(gcpSecret.Name),
+					)
+					Expect(err).ToNot(HaveOccurred())
+
+					trgt = ducktypes.WaitUntilReady(f.DynamicClient, trgt)
+
+					trgtURL = ducktypes.Address(trgt)
+					Expect(trgtURL).ToNot(BeNil())
+				})
+			})
+
+			When("an event is sent to the target", func() {
+
+				BeforeEach(func() {
+					By("sending an event", func() {
+						sentEvent = e2ece.NewHelloEvent(f)
+
+						job := e2ece.RunEventSender(f.KubeClient, ns, trgtURL.String(), sentEvent)
+						apps.WaitForCompletion(f.KubeClient, job)
+					})
+				})
+
+				It("creates an object onto the bucket", func() {
+					var receivedObj []byte
+					var err error
+
+					By("listing the bucket objects", func() {
+						receivedObjs := e2estorage.GetObjectsReader(storageClient, bucketName)
+						Expect(receivedObjs).To(HaveLen(1),
+							"Received %d objects instead of 1", len(receivedObjs))
+
+						object := receivedObjs[0]
+						receivedObj, err = ioutil.ReadAll(object)
+						Expect(err).ToNot(HaveOccurred())
+					})
+
+					By("inspecting the object payload", func() {
+						Expect(receivedObj).To(Equal(sentEvent.Data()))
+					})
+				})
+			})
+		})
+	})
+
+	When("a client creates a target object with invalid specs", func() {
+
+		// Those tests do not require a real bucketName or gcpSecret
+		BeforeEach(func() {
+			bucketName = "test"
+			gcpSecret = &corev1.Secret{}
+		})
+
+		// Here we use
+		//   "Specify: the API server rejects ..., By: setting an invalid ..."
+		// instead of
+		//   "When: it sets an invalid ..., Specify: the API server rejects ..."
+		// to avoid creating a namespace for each spec, due to their simplicity.
+		Specify("the API server rejects the creation of that object", func() {
+
+			By("omitting the bucketName", func() {
```

Comment:
It would be nice to have a test which creates an object with an _invalid_ bucket name too, so we can exercise the regexp.

Reply:
I didn't add that test because the target doesn't have a regexp for the bucketName. My plan was to merge this PR and then update the target and the test to include it